### PR TITLE
do_cmake.sh: remove -DCMAKE_BUILD_TYPE=Debug from cmake options

### PR DESCRIPTION
**Pull request by tchaikov**
so we can use do_cmake.sh for building release builds, which are
required for performance tests.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: do_cmake.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ if type cmake3 > /dev/null 2>&1 ; then
 else
     CMAKE=cmake
 fi
-${CMAKE} -DCMAKE_BUILD_TYPE=Debug $ARGS "$@" .. || exit 1
+${CMAKE} $ARGS "$@" .. || exit 1
```
Comment (rzarzynski) [inline do_cmake.sh]: I'm afraid that outcome of  this change may be surprising to many people. Over the time developers might got used to and expect presence of the debug stuff in default-configured builds.
Can we satisfy the requirement for crimson perf testing in another way? If not, then we should consider sending prior notification to ceph-devel or, at least, print message signalizing the change at the end of `do_cmake.sh`, I think.

**Comment by rzarzynski** [comment]
@tchaikov: is this PR still relevant after merging https://github.com/ceph/ceph-build/pull/1405?

**Comment by tchaikov** [comment]
@rzarzynski i forgot this PR ! will follow it up in this week. thanks for the reminder!

File: do_cmake.sh (hunk 1 above)
Comment (tchaikov) [reply do_cmake.sh]: mail sent.

File: doc/install/build-ceph.rst [hunk 2]
```diff
@@ -32,10 +32,9 @@ repository and execute the following::
     cd build
     make
 
-.. note:: By default do_cmake.sh will build a debug version of ceph that may
-   perform up to 5 times slower with certain workloads. Pass 
-   '-DCMAKE_BUILD_TYPE=RelWithDebInfo' to do_cmake.sh if you would like to
-   build a release version of the ceph executables instead.
+.. note:: By default do_cmake.sh will build a release version of ceph with debug
+   symbols. Pass '-DCMAKE_BUILD_TYPE=Debug' to do_cmake.sh if you would like to
+   build a debug version of the ceph executables instead.
```
Comment (badone) [inline doc/install/build-ceph.rst]: There;s a similar passage in README.md

**Comment by badone** [comment]
the documentation changes should be minimal (if any) if https://github.com/ceph/ceph/pull/30799 merges

File: do_cmake.sh (hunk 1 above)
Comment (tchaikov) [reply do_cmake.sh]: with #30799, i think this change is backward compatible

File: doc/install/build-ceph.rst (hunk 2 above)
Comment (tchaikov) [reply doc/install/build-ceph.rst]: thanks, the document changes are dropped. as #30799 addresses the backward compatibility issue.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2019-10-09-1243/0dca79308d83afd02b99aa1aa8839d156215298c/

File: do_cmake.sh (hunk 1 above)
Comment (rzarzynski) [reply do_cmake.sh]: There shouldn't be a difference for the `perf test` bot as we already pass `-DCMAKE_BUILD_TYPE=Release` there.

Comment (tchaikov) [reply do_cmake.sh]: i see, didn't realize that cmake allows us to pass multiple `-D<FOO>=<BAR>` with the same `<FOO>`, and the last wins.

**Comment by tchaikov** [comment]
> @tchaikov: is this PR still relevant after merging [ceph/ceph-build#1405](https://github.com/ceph/ceph-build/pull/1405)? I think `-DCMAKE_BUILD_TYPE=Debug` gets override.

ahh, sorry, i missed this comment!

**Pull request merged**